### ISSUE TITLE
Allow to import operations prior to bank account creation in kresus

### ISSUE THREAD

**Issue by ZeHiro** [bnjbvr/kresus#190]
I plan to import old bank operations in kresus (before the creation of the account), but this is currently not possible : 
- The initial amount value is not dated (but could easily be by using the date of the youger operation, but what if an new imported operation as the same date as the current initial import date?

**Comment by bnjbvr** [comment]
I think that should be the exporter's responsibility to ensure that the format is the one expected by kresus (I don't want to pollute the import code with a lot of options). In that case, I guess you'd need to reset the initialAmount value:

`initialAmountForReal + sumOperationsPresentAtFirstImport = initialAmountInKresus`

so you need to subtract the total amount of the sum of operations present before (or the same day as) the first import. Does that make sense?

**Comment by ZeHiro** [comment]
Reseting the initialAmountValue could cause other problems : 
- Create account
- Import operations through weboob
- Import operations prior to account creation (with update of initial amount)
- Import other operations prior to account creation, but after previously imported operations.

More clean way to do that would be to add a "creationDate" to bank accounts and update diplay off balance : 
balance = initialAmoount
for operation in operations 
     if operation.date >= creationDate 
            balance = balance + operation.amount

For charts, the update of balance prior to cerationDate would be computed as 
amount [ date1] = initialAmount - sum of operations between date1 and creationDate

**Comment by bnjbvr** [comment]
Well, I don't like the idea to have to change the frontend just because we'd want to make importing older data easier. If the impact were very limited, that'd be fine; but we're talking changing already two places in the frontend for supporting something which ought to be an implementation detail on the server. Let's try to figure out something else.

Having the `creationDate` sounds like a good tradeoff. Every time we want to import operations, we'd do the following:

```
oldestDate = creationDate
for op in operations
  if op.date >= creationDate
    oldestDate = op.date
if oldestDate != creationDate
  adjust account balance by subtracting the total amount of all operations in [oldestDate, creationDate]
  update creationDate
```

This seems cleaner, what do you think?

**Comment by ZeHiro** [comment]
In a way it seems cleaner, but what if : 
- import data1 < RealcreationDate
- importe RealcreationDate > data2 > data1 ?

We need a new date :stuck_out_tongue_winking_eye:

**Comment by bnjbvr** [comment]
Ha, that's right. A few things to consider:

- I *really* don't want to pollute the frontend with these details. Sorry to be so pushing on that, but that really sounds like a detail that shouldn't be so closely correlated to the frontend, and I'd like the keep the components as loosely coupled as I can.
- Maybe the best tradeoff is a mix between my proposed way and the importer, who has to ensure that he never hits this case you've mentioned. So they'd need to ensure all imported data is older than the oldest operation.
- There's no way to import only operations at the moment, without importing a whole account, so that would either need to be implemented, or the importer would need to craft their own file by hand, mixing an export with older data.

All of this really looks not trivial. If you're willing to implement my former idea (or any better idea that'd do the equivalent without polluting the frontend), I'll be glad to review and merge your changes, though. Good luck!

**Comment by bnjbvr** [comment]
I've actually about a much simpler to do that: you could just implement #141 and get along with that, and only that. What do you think?

**Comment by ZeHiro** [comment]
Oh yeah! Nice :thumbup:

**Comment by ZeHiro** [comment]
If we go even further, it coult also include fix for #191 by 'just' including an 'update function for a given account, an update button would be added in the parameters view for each account.

**Comment by bnjbvr** [comment]
Trying to think out loud:

- The IBAN should be reloaded every time we update the accounts (if there's no IBAN field or it's different than the one saved in the db, (over)write it in db).
- The balance should be resync'd every time we **need** to, which, *hopefully*, should happen only in weird cases (kresus loses operations, or we're importing older operations).

So it seems the frequency at which we'd want to do the two operations would be very different, so although I like the idea to try to solve more problems at once, it doesn't seem appropriate in this particular case :) Thanks for the idea, though, it's good to think about it!

**Comment by ZeHiro** [comment]
In fact the question here was more, do we integrate a full resync button, including also resync of amount (and IBAN...). Or only the resync of amount?

**Comment by bnjbvr** [comment]
Right, I didn't make myself clear, sorry :) My answer to this question was to use a different button for the resync of amount:

- the IBAN would be resync'd every time we resync all accounts (I plan to allow automating accounts resyncing, but that's another story)
- there could be a parameters button in front of each bank account, which would open a modal; on this modal, one could manually set the balance amount, or resync it with the bank's balance

**Comment by ZeHiro** [comment]
Ok. I'll start to work on this one next week. After the operation type selection.

**Comment by nikaro** [comment]
This feature would be nice to have.

Currently, i don't know why, my balances are wrong (#415) and i don't know how to fix it without deleting/re-adding my accounts.
